ops/services/app_functions/report_stream_publisher/functions/SingleTestEventPublisher/index.js

```diff
@@ -0,0 +1,120 @@
+const https = require('https');
+const uploaderVersion = "2021-04-26";
+
+module.exports = async function(context, message) {
+  const response = JSON.parse(await postRequest(
+      fetchFromEnvironmentOrThrow(
+          "REPORT_STREAM_URL",
+          "ReportStream URL to which tests should be reported"
+      ),
+      {
+        'x-functions-key': fetchFromEnvironmentOrThrow(
+            "REPORT_STREAM_TOKEN",
+            "ReportStream token"
+        ),
+        'content-type': 'text/csv',
+        'client': 'simple_report_queue_worker',
+        'x-api-version': uploaderVersion,
+      },
+      jsonToCsv(message)
+  ));
+
+  logResponseData(response.errors, err => {
+    console.error("Upload unsuccessful", err);
+    throw new Error("Upload did not complete successfully");
+  });
+  logResponseData(response.warnings, el => console.warn("Warning received", el));
+  logResponseData(response.destinations, el => console.log("Upload succeeded", el));
+};
+
+function logResponseData(responseDataArray, logFunc) {
+  if (Array.isArray(responseDataArray) && responseDataArray.length > 0) {
+    for (const el of responseDataArray) {
+      logFunc(el);
+    }
+  }
+}
+
+/**
+ * @param url The URL to which data should be sent.
+ * @param headers Custom headers to attach to the request.
+ * @param payload The request payload
+ * @returns {Promise<string>}
+ */
+function postRequest(url, headers, payload) {
+  return new Promise((resolve, reject) => {
+    const {
+      hostname,
+      pathname,
+      port,
+      protocol,
+      search,
+    } = new URL(url);
+    const requestOptions = {
+      headers,
+      hostname,
+      method: "POST",
+      path: pathname + search,
+      port: port ? Number(port) : undefined,
+      protocol,
+    };
+
+    const request = https.request(requestOptions, res => {
```

Comment:
Reading this, my first instinct would be to suggest adding a package like axios to deal with this low-level stream interaction, bc it will be easier to maintain in the long run

But, in the case of a utility running as a FaaS like this, I suppose it makes sense to keep it to zero dependencies for now